app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -0,0 +1,404 @@
+        }
+
+        override fun onRouteAlternativesError(error: RouteAlternativesError) {
+            // no impl
+        }
+    }
+
+    /**
+     * Gets notified with progress along the currently active route.
+     */
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeLineApi.updateWithRouteProgress(routeProgress) { result ->
+            binding.mapView.getMapboxMap().getStyle()?.apply {
+                routeLineView.renderRouteLineUpdate(this, result)
+            }
+        }
+    }
+
+    /**
+     * This is one way to keep the route(s) appearing on the map in sync with
+     * MapboxNavigation. When this observer is called the route data is used to draw route(s)
+     * on the map.
+     */
+    private val routesObserver = RoutesObserver { result ->
+        lifecycleScope.launch {
+            routeLineApi.setNavigationRoutes(
+                newRoutes = result.navigationRoutes,
+                alternativeRoutesMetadata = mapboxNavigation.getAlternativeMetadataFor(
+                    result.navigationRoutes
+                )
+            ).apply {
+                routeLineView.renderRouteDrawData(
+                    binding.mapView.getMapboxMap().getStyle()!!,
+                    this
+                )
+            }
+        }
+    }
+
+    /**
+     * Click on any point of the alternative route on the map to make it primary.
+     */
+    private val mapClickListener = OnMapClickListener {
+        lifecycleScope.launch {
+            routeLineApi.findClosestRoute(
+                it,
+                binding.mapView.getMapboxMap(),
+                routeClickPadding
+            ) {
+                val routeFound = it.value?.navigationRoute
+                if (routeFound != null && routeFound != routeLineApi.getPrimaryNavigationRoute()) {
+                    val reOrderedRoutes = routeLineApi.getNavigationRoutes()
+                        .filter { it != routeFound }
+                        .toMutableList()
+                        .also {
+                            it.add(0, routeFound)
+                        }
+                    mapboxNavigation.setNavigationRoutes(reOrderedRoutes)
+                }
+            }
+        }
+        false
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityShowAlternativeRoutesBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // initialize Mapbox Navigation
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(this)
+                .accessToken(getString(R.string.mapbox_access_token))
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .routeAlternativesOptions(
+                    RouteAlternativesOptions.Builder()
+                        .intervalMillis(TimeUnit.SECONDS.toMillis(30))
+                        .build()
```

Comment:
AFAIK alternatives do a route request under the hood. Reducing the interval should significantly increase traffic usage. Should we warn users about it in the example as we use such a small interval? cc @RingerJK

Reply:
I don't have numbers, will try to find them

Reply:
it won't increase if we have some alternative